### DISCUSSION TITLE
Local package repositories with a specific set of packages?

### DISCUSSION THREAD

**Discussion by ZackArnold** [MiKTeX/miktex#713] [Q&A] [answered]
The [documentation ](https://miktex.org/howto/local-repository) lists the options for a package set for local package repositories. I don't see any definitions as to what packages that actually includes. 

Is there any possibility for defining a custom package set? For a trivial example, say I only wanted `xcolor` and its dependencies available through this repository. How would I do that without installing a full, multi-GB repository clone?

**Comment by edocevoli** [comment, marked as answer]
At the moment, each package has an associated "package set" attribute. This association is fixed and cannot be changed at the moment, i.e., currently it is not possible to define custom package sets.

**Reply by ZackArnold** [reply]
Is it possible to "trim" a package set, once downloaded as a local repository, to just the packages you require? For example, if I downloaded the complete set, and then trimmed it to 30 or 40 specific packages? Or would this result in lots of errors?

What I am curious about is being able to easily update and install packages from just a small list within a local repository.

**Reply by edocevoli** [reply]
Haven't tried it yet. But yes, I see no problem if you remove packages you never need.

**Reply by ZackArnold** [reply]
Ok, I can confirm that this does work. It seems if I do try to request a package that isn't in the repository I get an error, which is what I expected. I can work with this. It was easy enough to make a script to collect the require archives from a CTAN mirror.

Is there anyway to specify this local repository in the form of a local server URL? 
For example, if I wanted to point MiKTeX console to a local server hosting the repository as opposed to, say, a network share?

**Reply by edocevoli** [reply]
No, that's not possible at the moment. IIRC, there is enhancement issue ticket for the feature.

**Reply by ZackArnold** [reply]
Ok. Thanks for your insight!
